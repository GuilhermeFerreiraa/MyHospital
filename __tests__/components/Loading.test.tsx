import React from 'react';
import Loading from '@/components/loading';
import colors from 'tailwindcss/colors';

const ReactTestRenderer = require('react-test-renderer');

describe('Loading Component', () => {
	it('renders correctly', () => {
		const tree = ReactTestRenderer.create(<Loading />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('contains an ActivityIndicator with correct properties', () => {
		const component = ReactTestRenderer.create(<Loading />);
		const instance = component.root;

		const activityIndicator = instance.findByType('ActivityIndicator');
		expect(activityIndicator.props.color).toBe(colors.white);
		expect(activityIndicator.props.size).toBe(24);
	});
});
