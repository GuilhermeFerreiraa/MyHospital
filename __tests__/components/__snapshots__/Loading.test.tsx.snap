// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component renders correctly 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "position": "absolute",
      },
      {
        "width": "100%",
      },
      {
        "height": "100%",
      },
      {
        "opacity": 0.75,
      },
      {
        "zIndex": 50,
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "backgroundColor": "#23b0ba",
      },
    ]
  }
  testID="loading-view"
>
  <ActivityIndicator
    color="#fff"
    size={24}
    testID="loading-indicator"
  />
</View>
`;
